lib/zlib.js

```diff
@@ -379,9 +379,15 @@ function Zlib(opts, mode) {
   var strategy = exports.Z_DEFAULT_STRATEGY;
   if (typeof opts.strategy === 'number') strategy = opts.strategy;
 
-  this._handle.init(opts.windowBits || exports.Z_DEFAULT_WINDOWBITS,
+  var windowBits = exports.Z_DEFAULT_WINDOWBITS;
+  if (typeof opts.windowBits === 'number') windowBits = opts.windowBits;
+
+  var memLevel = exports.Z_DEFAULT_MEMLEVEL;
+  if (typeof opts.memLevel === 'number') memLevel = opts.memLevel;
+
+  this._handle.init(windowBits || exports.Z_DEFAULT_WINDOWBITS,
```

Comment:
If your changing this line anyway, IMHO handle the fallback in 383 (`opts.windowBits && typeof...`)